### PR TITLE
Fix/categories block label missing

### PR DESCRIPTION
**Pull request by Luciaisacomputer**
## Description

The categories block does not have a label when it is set to "Display as dropdown" as mentioned in #10871. This updates the markup in the admin to render a visually hidden label for assistive technology to provide context to the dropdown. 

Also mentioned in the related issue above is that the labels should possibly be visual by default. Once merged, another PR can be opened to address that.

One thing to note is that various screen readers will respond differently to select dropdowns based on the type of label used (implicit vs explicit) so even though in my testing it never read the label because I was using VoiceOver and Safari, other browsers/screen readers will respond differently
(http://www.davidmacd.com/blog/select-box-label.html)

## How has this been tested?
Tested on OSX in Chrome, Safari, Firefox, Edge,  and IE11 using Browserstack. I used Voice Over + Safari to test that the label is properly read. 

## Screenshots 
screen shot 2018-10-22 at 2 22 38 pm

screen shot 2018-10-22 at 2 24 40 pm

(My category name was test, which I now realize should probably be something a little more contextual)

## Types of changes
Bug fix

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. 
- [X] My code follows the accessibility standards. 
- [X] My code has proper inline documentation.

### CODE REVIEW

File: packages/block-library/src/categories/edit.js [hunk 1]
```diff
@@ -106,14 +107,18 @@ class CategoriesEdit extends Component {
...
-
 		return (
-			<select className={ `${ this.props.className }__dropdown` }>
-				{ categories.map( ( category ) => this.renderCategoryDropdownItem( category, 0 ) ) }
-			</select>
+			<Fragment>
+				<label htmlFor={ `${ className }-${ instanceId }` } className="screen-reader-text">
+					{ __( 'Categories' ) }
+				</label>
+				<select id={ `${ className }-${ instanceId }` } className={ `${ className }__dropdown` }>
```
Comment (gziolo) [inline packages/block-library/src/categories/edit.js]: I think `id` can be statically set in here to something like:
```js
const selectId = 'block-library-category';
...
<label htmlFor={ `${ selectId }-${ instanceId }` } className="screen-reader-text">
....
<select id={ `${ selectId }-${ instanceId }` } className={ `${ className }__dropdown` }>

Comment (LukePettway) [reply packages/block-library/src/categories/edit.js]: Thanks fora the feedback, I'll make this change and push a commit when I find some time later today.

File: packages/block-library/src/categories/edit.js [hunk 2]
```diff
@@ -1,11 +1,16 @@
+/**
```
Comment (gziolo) [inline packages/block-library/src/categories/edit.js]: I noticed that `loadash` was located in the wrong section of imports. It's fixed now.

File: packages/block-library/src/categories/edit.js [hunk 3]
```diff
@@ -106,14 +111,19 @@ class CategoriesEdit extends Component {
 	}
 
 	renderCategoryDropdown() {
-		const { showHierarchy } = this.props.attributes;
+		const { showHierarchy, instanceId, className } = this.props;
 		const parentId = showHierarchy ? 0 : null;
 		const categories = this.getCategories( parentId );
-
+		const selectId = `blocks-category-select-${ instanceId }`;
```
Comment (gziolo) [inline packages/block-library/src/categories/edit.js]: I updated this id myself to align with other similar ids added in different places.

**Review by gziolo — APPROVED** [review]
I applied to minor fixes. It's goo to go. Thank you for working on this.

**Comment by tofumatt** [comment]
I agree, we should file another issue for it; right now it just being a dropdown in the middle of content really lacks context for *all* users. Filed: #11002 and #11003.

File: packages/block-library/src/categories/edit.js (hunk 2 above)
Comment (LukePettway) [reply packages/block-library/src/categories/edit.js]: Thanks! I wasn't sure what order to put it in I'll make sure to check out the documentation for the JS standards

**Comment by youknowriad** [comment]
The issue also mentions that we should show this label on the frontend which is not the case with the PR. Any taker for a follow-up PR for the PHP side of things?

**Comment by joedolson** [comment]
This seems to have been lost in the current trunk; there is no longer a label associated with the category block dropdown.

**Comment by youknowriad** [comment]
@joedolson would you mind opening a new issue for this in order for us to debug it properly.

**Comment by youknowriad** [comment]
Looking at the diff above, this adds the label in the editor but not in the frontend. I guess it was never fixed in the frontend.

**Pull request merged**